chain/network/src/peer_manager/connection/mod.rs

```diff
@@ -158,6 +163,8 @@ pub(crate) struct PoolSnapshot {
     /// Connections which have completed the handshake and are ready
     /// for transmitting messages.
     pub ready: im::HashMap<PeerId, Arc<Connection>>,
+    /// index on `ready` by Connection.owned_account.account_key.
+    pub ready_by_account_key: im::HashMap<PublicKey, Arc<Connection>>,
```

Comment:
I'm thinking whether the value should be a single connection or a map of connections.. 

let's discuss this.

Reply:
if there is more than one node handling a specific account_key it means that it is a malicious node  (with our current implementation of the node). It also means that we don't even know to which of these nodes we should route the messages to which which will make our system even harder to debug realtime. Also selecting just one of these nodes is a consistent behavior with AnnounceAccount broadcasts - we keep only 1 announce per account_id.

Reply:
discussed offline. It is possible that after a validator node crash, a TCP zombie connection will still be in ESTABLISHED state for possibly a long time, which will effectively block the validator node from reconnecting after restart. It is not trivial to work around, I've added an extensive description in a comment and added logging to properly observe such a situation. We have a fallback to TIER2 transmission in case of TIER1 connectivity problems, so the situation should be fine for now.